Providers/Modules/Plugins/Common/plugin/oms_configuration.rb

```diff
@@ -232,7 +274,13 @@ def load_configuration(conf_path, cert_path, key_path)
           OMS::Log.error_once("Error loading certs: #{e}")
           return false
         end
-
+    
+        @@AzureRegion = get_azure_region_from_imds()
```

Comment:
is it fine to do this without a thread like the resource id?

Reply:
yes region wont change